### PR TITLE
Bug 1094202  Implement the Share To Extension UI

### PR DESCRIPTION
**Pull request by st3fan**
This patch brings the Share To extension UI up to date with the latest specification. There are a couple of items though:

* I am reusing icons (login in navigation bar, bookmark, add to reading list) from the application icon set and the reading list icon set. I can either scale these down if needed, or we can incorporate new sizes for this dialog.
* I've left some of the fonts to the system default. Specifically the navigation bar buttons and the table row label. I don't know if we should actually change the size of those.
* Note that the dialog has a dynamic size. It's maximum width is 380px. But it also enforces a 16px padding left and right. So on small width devices like the 4S and 5, the max width is 288px. (This is done with auto layout rules, so we can change these dynamics)

Tested on iPhone 4S, iPhone 5, 6, 6 Plus and iPad. Fits & looks good on all these device sizes.

This is not pixel/spec perfect, but it is easy to change paddings, font sizes, colors and item sizes by editing the `ShareViewController.swift` file and changing the settings in the `ShareDialogControllerUX` struct.

### CODE REVIEW

File: Extensions/ShareTo/ShareViewController.swift [hunk 1]
```diff
@@ -127,36 +157,30 @@ class ShareDialogController: UIViewController, UITableViewDataSource, UITableVie
...
+            "V:[link]-\(ShareDialogControllerUX.ItemLinkBottomPadding)-[divider]",
+
+            "H:|[table]|",
+            "V:[divider][table]",
+            "V:[table(\(ShareDialogControllerUX.TableHeight))]|"
+        ]
+
+        for constraint in constraints {
+            view.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat(constraint, options: NSLayoutFormatOptions.allZeros, metrics: nil, views: views))
+        }
```
Comment (wesj) [inline Extensions/ShareTo/ShareViewController.swift]: Should this all be in an updateConstraints method?

File: Extensions/ShareTo/ShareViewController.swift [hunk 2]
```diff
@@ -25,6 +25,38 @@ protocol ShareControllerDelegate {
...
+private struct ShareDialogControllerUX {
+    static let CornerRadius: CGFloat = 4                                                            // Corner radius of the dialog
+
+    static let NavigationBarTintColor = UIColor(rgb: 0xf37c00)                                      // Tint color changes the text color in the navigation bar
+    static let NavigationBarCancelButtonFont = UIFont.systemFontOfSize(UIFont.buttonFontSize())     // System default
+    static let NavigationBarAddButtonFont = UIFont.boldSystemFontOfSize(UIFont.buttonFontSize())    // System default
+    static let NavigationBarIconSize = 32                                                           // Width and height of the icon
+    static let NavigationBarBottomPadding = 12
+
+    static let ItemTitleFont = UIFont(name: "HelveticaNeue-Medium", size: 12)
```
Comment (wesj) [inline Extensions/ShareTo/ShareViewController.swift]: Are we comfortable using System fonts for the Cancel/Add buttons and hardcoded ones here? Same for sizes?

File: Extensions/ShareTo/ShareViewController.swift (hunk 1 above)
Comment (st3fan) [inline Extensions/ShareTo/ShareViewController.swift]: I'll refactor it when I move it all to use Snap.

File: Extensions/ShareTo/ShareViewController.swift (hunk 2 above)
Comment (st3fan) [inline Extensions/ShareTo/ShareViewController.swift]: We've made the decision to use Helvetica Neue everywhere, which is the same as the system font. However, we sometimes need variants like *Medium* or an odd point size for which there is no `UIFont` convenience method.

I agree that this is confusing and I am not entirely sure how to deal with this.

**Pull request merged**